cmd/fpga_plugin/fpga_plugin.go

```diff
@@ -172,17 +175,36 @@ type devicePlugin struct {
 	ignoreAfuIDs       bool
 	ignoreEmptyRegions bool
 	annotationValue    string
+
+	scanTicker *time.Ticker
+	scanDone   chan bool
 }
 
 // newDevicePlugin returns new instance of devicePlugin
-func newDevicePlugin(mode string) (*devicePlugin, error) {
-	if _, err := os.Stat(sysfsDirectoryOPAE); os.IsNotExist(err) {
-		if _, err = os.Stat(sysfsDirectoryDFL); os.IsNotExist(err) {
-			return nil, fmt.Errorf("kernel driver is not loaded: neither %s nor %s sysfs entry exists", sysfsDirectoryOPAE, sysfsDirectoryDFL)
+func newDevicePlugin(mode string, rootPath string) (*devicePlugin, error) {
+	var dp *devicePlugin
+	var err error
+
+	sysfsPathOPAE := path.Join(rootPath, sysfsDirectoryOPAE)
+	devfsPath := path.Join(rootPath, devfsDirectory)
+	if _, err := os.Stat(sysfsPathOPAE); os.IsNotExist(err) {
```

Comment:
Let's remove this colon to lessen ambiguity, because I'm having hard time to understand if `err` in the line 195 is from the function's scope of from this block's scope.

Reply:
done. Thanks for pointing this out to me.